pkg/storage/stores/shipper/compactor/compactor.go

```diff
@@ -229,15 +229,20 @@ func (c *Compactor) init(storageConfig storage.Config, schemaConfig config.Schem
 			return err
 		}
 
-		if c.deleteMode == deletion.WholeStreamDeletion {
+		if c.deleteMode == deletion.WholeStreamDeletion || c.deleteMode == deletion.FilterAndDelete {
```

Comment:
I think we need this for any mode (except disabled) so the cache invalidation can work in `FilterOnly` mode

Reply:
Updated to use the switch statement as in https://github.com/grafana/loki/pull/5922/files